data-collection/src/main/java/refactoringml/ProcessMetricsCollector.java

```diff
@@ -89,11 +89,10 @@ public void collect() throws IOException {
 					try {
 						db.openSession();
 						refactoredClasses = collectProcessMetricsOfRefactoredCommit(commit);
-						db.commit();
 					} catch (Exception e) {
 						log.error("Error when collecting process metrics in commit " + commit.getName(), e);
-						db.rollback();
 					} finally {
+						db.commit();
```

Comment:
The commit has to be inside of the `try` block. It can throw an exception! The way it is now, if it throws an exception, the `close` will never be invoked